File: app/src/main/java/ua/com/foxminded/locationtrackera/background/LocationServiceContract.java
package ua.com.foxminded.locationtrackera.background;

import io.reactivex.rxjava3.annotations.NonNull;
import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Observable;
import ua.com.foxminded.locationtrackera.model.locations.UserLocation;

public interface LocationServiceContract {

    interface Presenter {

        void onStart();

        @NonNull Completable saveUserLocation(UserLocation location);

        void onDestroy();

        Observable<Integer> getGpsStatusObservable();
    }
}

File: app/src/main/java/ua/com/foxminded/locationtrackera/background/LocationServiceContract.kt
package ua.com.foxminded.locationtrackera.background

import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Observable
import ua.com.foxminded.locationtrackera.models.locations.UserLocation

interface LocationServiceContract {

    interface Presenter {

        fun onStart()

        fun saveUserLocation(location: UserLocation): Completable

        fun onDestroy()

        val gpsStatusObservable: Observable<Int>
    }
}
